File: python/435_无重叠区间_medium.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if not intervals:
            return 0
        intervals.sort(key=lambda x:x[1])
        end = intervals[0][1]
        ans = 0
        for interval in intervals[1:]:
            start = interval[0]
            if start < end:
                ans += 1
            else:
                end = interval[1]
        return ans

File: cpp/435_无重叠区间_medium.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int N = intervals.size();
        if(!N)
            return 0;
        sort(intervals.begin(), intervals.end(), [](const vector<int> &a, const vector<int> &b){
            return a[1] < b[1];
        });
        int end = intervals[0][1];
        int ans = 0;
        for(int i=1; i<N; i++){
            int start = intervals[i][0];
            if(start < end)
                ans++;
            else
                end = intervals[i][1];
        }
        return ans;
    }
};